glean-core/src/metrics/mod.rs

```diff
@@ -165,7 +169,40 @@ pub trait MetricType {
     /// This depends on the metrics own state, as determined by its metadata,
     /// and whether upload is enabled on the Glean object.
     fn should_record(&self, glean: &Glean) -> bool {
-        glean.is_upload_enabled() && self.meta().should_record()
+        if !glean.is_upload_enabled() {
+            return false;
+        }
+
+        // Get the current disabled field from the metric metadata, including
+        // the encoded Nimbus epoch
+        let disabled_field = self.meta().disabled.load(Ordering::Relaxed);
+        // Grab the epoch from the upper nibble
+        let epoch = disabled_field >> 4;
+        // Get the disabled flag from the lower nibble
+        let disabled = disabled_field & 0xF;
+        // Get the current Nimbus epoch to see if we need to bother with the
+        // more expensive HashMap lookup
+        let nimbus_epoch = glean.nimbus_epoch.load(Ordering::Acquire);
+        if epoch == nimbus_epoch {
+            return disabled == 0;
+        }
+        // The epoch's didn't match so we need to look up the disabled flag
+        // by the base_identifier from the in-memory HashMap
+        let metrics_disabled = &glean.nimbus_metrics_config.lock().unwrap().metrics_disabled;
+        // Get the value from the remote configuration if it is there, otherwise return the default value.
+        let current_disabled =
+            if let Some(is_disabled) = metrics_disabled.get(&self.meta().base_identifier()) {
+                u8::from(*is_disabled)
+            } else {
+                u8::from(self.meta().defaults.disabled)
+            };
+
+        // Re-encode the epoch and enabled status and update the metadata
+        let new_disabled = (nimbus_epoch << 4) | (current_disabled & 0xF);
+        self.meta().disabled.store(new_disabled, Ordering::Relaxed);
```

Comment:
So _technically_ nothing stops multiple `should_record()` to run in parallel, and both reading `self.meta().disabled`, then later store to it again.
In between it can read `nimbus_metrics_config`, which itself _could_ be modified in between those 2 reads.
Now we could write the _wrong_ `nimbus_epoch | current_disabled` value until at least the next time this metric is written.
All in all at worst we would see that metric enabled/disabled wrongly once. 

But also from the outside everything is tunneled through the dispatcher, so this should never ever happen to begin with.
I therefore say this is all fine, but maybe we can leave a comment on top calling out that it is fine that this reading/calculating/writing is not atomic?